prime-router/src/main/kotlin/history/db/ReportGraph.kt

```diff
@@ -0,0 +1,100 @@
+package gov.cdc.prime.router.history.db
+
+import gov.cdc.prime.router.azure.DatabaseAccess
+import gov.cdc.prime.router.azure.db.Tables.COVID_RESULT_METADATA
+import gov.cdc.prime.router.azure.db.Tables.REPORT_FILE
+import gov.cdc.prime.router.azure.db.Tables.REPORT_LINEAGE
+import gov.cdc.prime.router.azure.db.tables.pojos.CovidResultMetadata
+import gov.cdc.prime.router.common.BaseEngine
+import org.apache.logging.log4j.kotlin.Logging
+import org.jooq.CommonTableExpression
+import org.jooq.Record
+import org.jooq.Record2
+import org.jooq.impl.DSL
+import org.jooq.impl.DSL.selectDistinct
+import org.jooq.impl.SQLDataType
+import java.util.UUID
+
+class ReportGraph(
+    val db: DatabaseAccess = BaseEngine.databaseAccessSingleton
+) : Logging {
+
+    /**
+     * Returns all the metadata for the items in the past in reports; will recursively walk up the report lineage
+     * and then filter down to reports where the sender is set in order to find the metadata rows
+     *
+     * @param childReportIds the list of ids to start from
+     *
+     */
+    fun getMetadataForReports(childReportIds: List<UUID>): List<CovidResultMetadata> {
+        val lineage = lineageCte(childReportIds)
+
+        val sourceReportIds =
+            sourceReportsCte(lineage)
+
+        val metadata = metadataExpression(sourceReportIds)
+
+        return db.transactReturning { txn ->
+            DSL.using(txn)
+                .withRecursive(lineage)
+                .with(sourceReportIds)
+                .with(metadata)
+                .selectDistinct(metadata.asterisk())
```

Comment:
This provides an example of how the CTEs can be combined to get different data out of the graph

Reply:
Why does the `sourceReportsCte` have "Cte" in the name but the others don't, like `ancestorLineageExpression`? We should be consistent in our naming conventions. I don't have a preference either way, but if you explain that this class is built around CTEs in the class level Kdoc then my preference is to not include "cte" in the names.

Reply:
Gave a consistent name of `*CommonTableExpression`